storage/plugins/archives/RicJsonLdBuilder.php

```
@@ -0,0 +1,1399 @@
+        // dates) and Phase-1 buildDateRange (xsd:gYear for year-only)
+        // accept the same VARCHAR(20) shape. Detect year-only values
+        // (e.g. "1789") and emit them as xsd:gYear; full ISO dates
+        // (e.g. "1789-07-14") remain xsd:date. A bare year is NOT a
+        // valid xsd:date lexical value, so this avoids producing
+        // RDF that fails XSD validation.
+        $start = $this->str($activity, 'date_start');
+        $end   = $this->str($activity, 'date_end');
+        if ($start !== '' || $end !== '') {
+            $dateNode = ['@type' => 'ric:DateRange'];
+            if ($start !== '') {
+                $dt = preg_match('/^\d{4}$/', $start) === 1 ? 'xsd:gYear' : 'xsd:date';
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $dt. Configured minimum length is 3. (undefined)

(ShortVariable)